Backend/app/core/tts.py

```
@@ -0,0 +1,307 @@
+        session_id: Optional[str] = None
+    ) -> Dict[str, str]:
+        if character_id not in self.character_voices:
+            logger.error(f"❌ Unknown character voice: {character_id}")
+            return {
+                "success": False,
+                "error": f"Unknown character: {character_id}",
+                "character_id": character_id
+            }
+        
+        if not text or not text.strip():
+            logger.error(f"❌ Empty text provided for TTS")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)